src/apps/Altinn.AccessManagement/src/Altinn.AccessManagement.Api.Internal/Controllers/SystemUserClientDelegationController.cs

```diff
@@ -0,0 +1,254 @@
+        if (userId == Guid.Empty)
+        {
+            return Unauthorized();
+        }
+
+        if (roles != null && roles.Length > 0)
+        {
+            var invalidRoles = roles.Where(role => !validClientRoles.Contains(role));
+            if (invalidRoles.Any())
+            {
+                return BadRequest($"Invalid role filter: '{string.Join(",", invalidRoles)}'. Valid Client roles are: '{string.Join(", ", validClientRoles)}'");
+            }
+        }
+        else
+        {
+            roles = validClientRoles;
+        }
+
+        if (packages == null || packages.Length == 0)
+        {
+            packages = [];
+        }
+
+        var clients = await assignmentService.GetClients(party, roles, packages, cancellationToken);
+
+        return Ok(clients);
+    }
+
+    /// <summary>
+    /// Gets all client delegations for a given system user
+    /// </summary>
+    /// <param name="party">The party the authenticated user is performing client administration on behalf of</param>
+    /// <param name="systemUser">The system user the authenticated user is delegating access to</param>
+    /// <param name="cancellationToken">The <see cref="CancellationToken"/></param>
+    /// <returns><seealso cref="SystemUserClientConnectionDto"/>List of connections</returns>
+    [HttpGet]
+    [Authorize(Policy = AuthzConstants.SCOPE_ENDUSER_CLIENTDELEGATION_READ)]
+    [Authorize(Policy = AuthzConstants.POLICY_CLIENTDELEGATION_READ)]
+    public async Task<ActionResult<IEnumerable<SystemUserClientConnectionDto>>> GetClientDelegations([FromQuery] Guid party, [FromQuery] Guid systemUser, CancellationToken cancellationToken = default)
+    {
+        var userId = AuthenticationHelper.GetPartyUuid(HttpContext);
+        if (userId == Guid.Empty)
+        {
+            return Unauthorized();
+        }
+
+        return Ok(await connectionService.GetConnectionsToAgent(viaId: party, toId: systemUser, cancellationToken: cancellationToken));
+    }
+
+    /// <summary>
+    /// Post client delegation
+    /// </summary>
+    /// <param name="party">The party the authenticated user is performing client administration on behalf of</param>
+    /// <param name="request">Request Dto</param>
+    /// <param name="cancellationToken">The <see cref="CancellationToken"/></param>
+    /// <returns><seealso cref="CreateDelegationResponseDto"/>List of delegation responses</returns>
+    [HttpPost]
+    [Authorize(Policy = AuthzConstants.SCOPE_ENDUSER_CLIENTDELEGATION_WRITE)]
+    [Authorize(Policy = AuthzConstants.POLICY_CLIENTDELEGATION_WRITE)]
+    [AuditJWTClaimToDb(Claim = AltinnCoreClaimTypes.PartyUuid, System = AuditDefaults.InternalApiStr)]
+    public async Task<ActionResult<IEnumerable<CreateDelegationResponseDto>>> PostClientDelegation([FromQuery] Guid party, [FromBody] CreateSystemDelegationRequestDto request, CancellationToken cancellationToken = default)
+    {
+        var userId = AuthenticationHelper.GetPartyUuid(HttpContext);
```

Comment:
hva gjør denne?

Reply:
Copy paste fra forrige controller.
Helt enig, disse gjør ikke en dritt lenger. Unauthorized håndteres med Authorize attributtene.
På POST/DELETE måtte man ha disse før for å putte de på audit details manuelt men er nå også håndtert av AuditJWTClaimToDb.